File: src/main/java/br/com/ateliware/challenge/domain/GitHubRepository.java
package br.com.ateliware.challenge.domain;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import br.com.ateliware.challenge.type.Language;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class GitHubRepository {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long     id;
    @Column(length = 20480)
    private String   description;
    @Column(length = 2048)
    private String   homepage;
    private String   name;
    private String   fullName;
    private String   license;
    private int      forksCount;
    private int      stargazersCount;
    private int      watchersCount;
    private int      size;
    private int      openIssuesCount;
    private int      subscribersCount;
    private Date     pushedAt;
    private String   defaultBranch;
    private Language language;

}

File: backend/HiringChallenge.API/Domain/Models/GithubRepository.cs
using System;

namespace HiringChallenge.API.Domain.Models
{
    public class GithubRepository
    {
        public GithubRepository()
        {
            Id = Guid.NewGuid();
        }

        public Guid Id { get; set; }
        public long RepositoryId { get; set; }
        public string Name { get; set; }
        public string FullName { get; set; }
        public string Owner { get; set; }
        public string HtmlUrl { get; set; }
        public string Description { get; set; }
        public string Language { get; set; }
        public int StargazersCount { get; set; }
        public DateTime CreationDate { get; set; }
    }
}